src/Raven.Server/Documents/Queries/LuceneQueryHelper.cs

```diff
@@ -358,7 +358,7 @@ private static Query CreateRange(Index index, string fieldName, double minValue,
 
         private static Query MaybeCacheQuery(Index index, Query query)
         {
-            if (index.Configuration.QueryClauseCacheDisabled)
+            if (index is null || index.Configuration.QueryClauseCacheDisabled)
```

Comment:
How can the `index` be null?

Reply:
During building Lucene query: 
https://github.com/ravendb/ravendb/blob/611a8edbd80356c82e2438aec76ada4c337fafee/src/Raven.Server/Documents/Queries/QueryBuilder.cs#L414

And when we look into the methods:
https://github.com/ravendb/ravendb/blob/611a8edbd80356c82e2438aec76ada4c337fafee/src/Raven.Server/Documents/Queries/QueryBuilder.cs#L924

Reply:
That actually executes code? because it may mean that we need to check query caching + boosting

Reply:
You specified explicitly `index: null` so we though that it's intentional to _not_ apply this optimization. This applies to boosting, fuzzy, proximity ...

Reply:
I don't think I ever thought we'll reach that portion of the code. It was just to fix the API, It think.

Reply:
Opened https://github.com/ravendb/ravendb/pull/13455